mscore/prefsdialog.cpp

```diff
@@ -1007,130 +1302,95 @@ void PreferenceDialog::apply()
-    } else if (newSession->isChecked()) {
-        preferences.setCustomPreference<SessionStart>(PREF_APP_STARTUP_SESSIONSTART, SessionStart::NEW);
-    } else if (scoreSession->isChecked()) {
-        preferences.setCustomPreference<SessionStart>(PREF_APP_STARTUP_SESSIONSTART, SessionStart::SCORE);
-    } else if (emptySession->isChecked()) {
-        preferences.setCustomPreference<SessionStart>(PREF_APP_STARTUP_SESSIONSTART, SessionStart::EMPTY);
-    }
-
-    preferences.setPreference(PREF_APP_AUTOSAVE_AUTOSAVETIME, autoSaveTime->value());
-    preferences.setPreference(PREF_APP_AUTOSAVE_USEAUTOSAVE, autoSave->isChecked());
-    preferences.setPreference(PREF_APP_PATHS_INSTRUMENTLIST1, instrumentList1->text());
-    preferences.setPreference(PREF_APP_PATHS_INSTRUMENTLIST2, instrumentList2->text());
-    preferences.setPreference(PREF_APP_PATHS_MYIMAGES, myImages->text());
-    preferences.setPreference(PREF_APP_PATHS_MYPLUGINS, myPlugins->text());
-    preferences.setPreference(PREF_APP_PATHS_MYSCORES, myScores->text());
-    preferences.setPreference(PREF_APP_PATHS_MYSOUNDFONTS, mySoundfonts->text());
-    preferences.setPreference(PREF_APP_PATHS_MYSTYLES, myStyles->text());
-    preferences.setPreference(PREF_APP_PATHS_MYTEMPLATES, myTemplates->text());
-    preferences.setPreference(PREF_APP_PATHS_MYEXTENSIONS, myExtensions->text());
-    preferences.setPreference(PREF_APP_STARTUP_STARTSCORE, sessionScore->text());
-    preferences.setPreference(PREF_EXPORT_AUDIO_SAMPLERATE, exportAudioSampleRate->currentData().toInt());
-    preferences.setPreference(PREF_EXPORT_MP3_BITRATE, exportMp3BitRate->currentData().toInt());
-    preferences.setPreference(PREF_EXPORT_MUSICXML_EXPORTLAYOUT, exportAllLayouts->isChecked());
-    preferences.setPreference(PREF_EXPORT_PDF_DPI, exportPdfDpi->value());
-    preferences.setPreference(PREF_EXPORT_PNG_RESOLUTION, pngResolution->value());
-    preferences.setPreference(PREF_EXPORT_PNG_USETRANSPARENCY, pngTransparent->isChecked());
-    preferences.setPreference(PREF_IMPORT_MUSICXML_IMPORTBREAKS, importBreaks->isChecked());
-    preferences.setPreference(PREF_IMPORT_MUSICXML_IMPORTLAYOUT, importLayout->isChecked());
-    if (resetElementPositionsAlwaysAsk->isChecked()) {
-        preferences.setPreference(PREF_IMPORT_COMPATIBILITY_RESET_ELEMENT_POSITIONS, "Ask");
-    } else if (resetElementPositionsYes->isChecked()) {
-        preferences.setPreference(PREF_IMPORT_COMPATIBILITY_RESET_ELEMENT_POSITIONS, "Yes");
-    } else if (resetElementPositionsNo->isChecked()) {
-        preferences.setPreference(PREF_IMPORT_COMPATIBILITY_RESET_ELEMENT_POSITIONS, "No");
+    QElapsedTimer timer;
+    timer.start();
+
+    if (buttonBox->button(QDialogButtonBox::Apply)->isEnabled() == false) {
+        return;
     }
+
+    bool uiStyleThemeChanged = false;
+    bool audioModified = false;
+
+    applySetActive(false);
+    buttonBox->button(QDialogButtonBox::Apply)->setText(tr("Applying..."));
```

Comment:
Please change to `Applying…`, i.e. with an ellipsis. See also #6201

Reply:
Done.